pkg/spiffe/spiffe.go

```diff
@@ -52,6 +53,34 @@ var (
 	spiffeLog = log.RegisterScope("spiffe", "SPIFFE library logging", 0)
 )
 
+type Identity struct {
+	TrustDomain    string
+	Namespace      string
+	ServiceAccount string
+}
+
+func ParseIdentity(s string) (Identity, error) {
+	if !strings.HasPrefix(s, URIPrefix) {
+		return Identity{}, fmt.Errorf("identity is not a spiffe format: %v", s)
+	}
+	split := strings.Split(s[9:], "/")
```

Comment:
9 is the len of URIPrefix? should not hardcode the length.

Reply:
Sscanf?